.agents/skills/gframework-doc-refresh/scripts/validate-code-blocks.sh

```
@@ -0,0 +1,56 @@
+
+LINE_NUMBER=0
+while IFS= read -r LINE; do
+    LINE_NUMBER=$((LINE_NUMBER + 1))
+
+    if echo "$LINE" | grep -qE '^```(cs|c#|C#)$'; then
+        echo "⚠ 警告: 第 $LINE_NUMBER 行使用了非标准 C# 标记，建议改为 csharp"
+        WARNING_COUNT=$((WARNING_COUNT + 1))
+    fi
+
+    if echo "$LINE" | grep -qE '^```$'; then
+        NEXT_LINE=$(sed -n "$((LINE_NUMBER + 1))p" "$FILE")
```

Shellcheck (0.11.0)

[info] Make sure not to read and write the same file in the same pipeline.

(SC2094)